sdk/digitaltwins/mgmt-v2020_12_01/src/test/java/com/microsoft/azure/management/digitaltwins/v2020_12_01/DigitalTwinsLifecycleTests.java

```diff
@@ -0,0 +1,132 @@
+package com.microsoft.azure.management.digitaltwins.v2020_12_01;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.arm.core.TestBase;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.CheckNameResultInner;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.DigitalTwinsDescriptionInner;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.DigitalTwinsManager;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.OperationInner;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.rest.RestClient;
+import com.microsoft.azure.management.resources.implementation.ResourceManager;
+import org.junit.Assert;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class DigitalTwinsLifecycleTests extends TestBase {
+
+    protected ResourceManager resourceManager;
+    protected DigitalTwinsManager digitalTwinsManager;
+    private static String rgName;
+    protected String domain;
+    private static String defaultInstanceName = "DigitalTwinsSdk";
+    private static String defaultRegion = Region.US_WEST_CENTRAL.toString();
+    private static String defaultResourceGroupName = "rg2b9842374ecf6";
+
+    @Override
+    protected void initializeClients(RestClient restClient, String defaultSubscription, String domain) throws IOException {
+        resourceManager = ResourceManager
+            .authenticate(restClient)
+            .withSubscription(defaultSubscription);
+
+        digitalTwinsManager = DigitalTwinsManager
+            .authenticate(restClient, defaultSubscription);
+
+        this.domain = domain;
+    }
+
+    @Test
+    public void lifecycleTest() throws InterruptedException {
+        //rgName = SdkContext.randomResourceName("rg", 20);
+        rgName = defaultResourceGroupName;
+
+        ResourceGroup group = resourceManager.resourceGroups()
+            .define(rgName)
+            .withRegion(defaultRegion)
+            .create();
+
+        Assert.assertNotNull(group);
+
+        try {
+            CheckNameResultInner checkNameResult = digitalTwinsManager.inner().digitalTwins()
+                .checkNameAvailability(defaultRegion, defaultInstanceName);
+
+            if (!checkNameResult.nameAvailable()) {
+                Iterator<DigitalTwinsDescription> allDigitalTwins = digitalTwinsManager.digitalTwins().listAsync().toBlocking().getIterator();
```

Comment:
Why list and delete all instances? This would make concurrent runs fail.

Reply:
we don't delete them all, we only delete the instance that is failing the name availability. on line 63 we are checking the name of the instance.